### PR TITLE
Add Perl

### PR DESCRIPTION
**Pull request by airencracken**
Adds support for benchmarking perl5

### CODE REVIEW

File: loops/perl/code.pl [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	die "Please input an integer\n";
+}
+
+my $r = int(rand(10000));
+my @a; #let perl allocate the array
+my $i;
+my $j;
+for $i (0..10000) {
+	for $j (0..100000) {
+		$a[$i] = $a[$i] + $j%$n;
```
Comment (zmughal) [inline loops/perl/code.pl:15]: This can also use `+=`.

File: loops/perl/code.pl [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	$n = int(shift @ARGV);
+} else {
+	die "Please input an integer\n";
+}
+
+my $r = int(rand(10000));
+my @a; #let perl allocate the array
+my $i;
+my $j;
+for $i (0..10000) {
```
Comment (zmughal) [inline loops/perl/code.pl:13]: `10_000-1`

File: loops/perl/code.pl [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	for $j (0..100000) {
```
Comment (zmughal) [inline loops/perl/code.pl:14]: `100_000-1`

File: loops/perl/code.pl [hunk 4]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/perl
+my $n;
+if(@ARGV) {
+	$n = int(shift @ARGV);
+} else {
+	die "Please input an integer\n";
+}
+
+my $r = int(rand(10000));
+my @a; #let perl allocate the array
```
Comment (zmughal) [inline loops/perl/code.pl:10]: To match the other implementations:

```perl
my @a = (0) x 10_000;
```

**Review by zmughal — CHANGES_REQUESTED** [review]

File: loops/perl/code.pl [hunk 5]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/perl
```
Comment (zmughal) [inline loops/perl/code.pl:1]: Maybe add

```perl
use strict;
use warnings;
```

here too?

Comment (airencracken) [reply loops/perl/code.pl:1]: Didn't see the need. There's a pointless warning for uninitialized variables. I could ignore it specifically, but meh.

File: loops/perl/code.pl (hunk 4 above)
Comment (airencracken) [reply loops/perl/code.pl:10]: Faster to let perl allocate it dynamically in my tests.

File: fibonacci/perl/code.pl [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+sub factorial {
+	my ($n, $total) = @_;
+
+	$total ||= 1;
+
+	return $total if ($n == 0);
+	@_ = ($n - 1, $n * $total);
+	goto &factorial;
+}
```
Comment (bddicken) [inline fibonacci/perl/code.pl:21]: This function needs to match the "naive" algorithm that the other implementations use. The idea is not to implement an optimally fast solution, but an equivalent solution.

File: run.sh [hunk 7]
```diff
@@ -25,3 +25,4 @@ run "PHP" "php ./php/code.php 40"
 run "R" "Rscript ./r/code.R 40"
 run "Python" "python3 ./py/code.py 40" 
 run "Dart" "./dart/code 40"
+run "Perl" "./perl/code.pl 40"
```
Comment (MuriloChianfa) [inline run.sh:28]: can we compile perl before run?

**Comment by airencracken** [comment]
Looks like someone else merged it.

**Pull request closed without merging**